MultiplayerMod/Multiplayer/Objects/MultiplayerObjects.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace MultiplayerMod.Multiplayer.Objects;
+
+public class MultiplayerObjects {
+
+    private readonly Dictionary<long, GameObject> objects = new();
+
+    public void Add(MultiplayerInstance instance) {
+        instance.Id ??= MultiplayerIdentityProvider.GetNextId();
+        objects[instance.Id.Value] = instance.gameObject;
+    }
+
+    public void Remove(long objectId) => objects.Remove(objectId);
+
+    public GameObject this[long id] => objects[id];
+
+    public void Clear() => objects.Clear();
```

Comment:
Should we call Clear upon hard sync? To avoid `objects` growing to big.

Reply:
Probably, it's still a draft.

Reply:
Added to hard sync.